src/azure-cli-core/azure/cli/core/breaking_change.py

```diff
@@ -0,0 +1,601 @@
+    It is recommended that the old name and the new name should be reserved in few releases.
+    :param target: name of the renamed command group, command or argument
+    :param new_name: new name
+    :param target_version: version where the breaking change is expected to happen.
+    :type target_version: TargetVersion
+    :param doc_link: link of the related document
+    """
+
+    def __init__(self, cmd, new_name, arg=None, target=None, target_version=NextBreakingChangeWindow(), doc_link=None):
+        super().__init__(cmd, arg, target, target_version)
+        self.new_name = new_name
+        self.doc_link = doc_link
+
+    @property
+    def message(self):
+        doc = self.format_doc_link(self.doc_link)
+        return f"'{self.target}' will be renamed to '{self.new_name}' {str(self._target_version)}.{doc}"
+
+
+class AzCLIOutputChange(BreakingChange):
+    """
+    The output of the command will be changed in a future release.
+    :param description: describe the changes in output
+    :param target_version: version where the breaking change is expected to happen.
+    :type target_version: TargetVersion
+    :param guide: how to adapt to the change
+    :param doc_link: link of the related document
+    """
+
+    def __init__(self, cmd, description: str, target_version=NextBreakingChangeWindow(), guide=None, doc_link=None):
+        super().__init__(cmd, None, None, target_version)
+        self.desc = description
+        self.guide = guide
+        self.doc_link = doc_link
+
+    @property
+    def message(self):
+        desc = self.desc.rstrip()
+        if desc and desc[-1] not in ',.;?!':
+            desc = desc + '.'
+        if self.guide:
+            guide = self.guide.rstrip()
+            if guide and guide[-1] not in ',.;?!':
+                guide = guide + '.'
+        else:
+            guide = ''
+        doc = self.format_doc_link(self.doc_link)
+        return f'The output will be changed {str(self.target_version)}. {desc} {guide}{doc}'
+
+
+class AzCLILogicChange(BreakingChange):
+    """
+    There would be a breaking change in the logic of the command in future release.
+    :param summary: a short summary about the breaking change
+    :param target_version: version where the breaking change is expected to happen.
+    :type target_version: TargetVersion
+    :param detail: detailed information
+    :param doc_link: link of the related document
+    """
+
+    def __init__(self, cmd, summary, target_version=NextBreakingChangeWindow(), detail=None, doc_link=None):
+        super().__init__(cmd, None, None, target_version)
+        self.summary = summary
+        self.detail = detail
+        self.doc_link = doc_link
+
+    @property
+    def message(self):
+        detail = f' {self.detail}' if self.detail else ''
+        return f'{self.summary} {str(self.target_version)}.{detail}{self.format_doc_link(self.doc_link)}'
```

Comment:
```suggestion
        return f'{self.summary} {str(self.target_version)}. {detail}{self.format_doc_link(self.doc_link)}'
```

Reply:
The `detail` variable contains a space at start. We don't need an additional space in formatted message to avoid extra space trailing.

Reply:
Got it, thanks~